app/models/training_module.rb

```diff
@@ -1,10 +1,43 @@
 class TrainingModule < YamlBase
-  set_filename 'training-modules'
+  set_filename ENV.fetch('TRAINING_MODULES', 'training-modules')
 
+  # Override basic behaviour so that root key is stored as name
   def self.load_file
-    # Override basic behaviour so that root key is stored as name
-    raw_data.map do |name, values|
-      values.merge(name: name)
+    raw_data.map { |name, values| values.merge(name: name) }
+  end
+
+  # @return [Array<TrainingModule>]
+  def self.by_state(user:, state:)
+    case state.to_sym
+    when :in_progress then all.select { |mod| user.in_progress?(mod) }
+    when :completed   then all.select { |mod| user.completed?(mod) }
+    when :upcoming    then all.select { |mod| user.upcoming?(mod) }[1..]
+    when :available   then Array(all.find { |mod| user.upcoming?(mod) })
     end
   end
+
+  # @return [Boolean]
+  def draft?
+    attributes.fetch(:draft, false)
+  end
+
+  # @return [Array<ModuleItem>]
+  def module_items
+    ModuleItem.where(training_module: name).to_a
+  end
+
+  # @return [Array<ModuleItem>]
+  def module_items_by_type(type)
+    ModuleItem.where(training_module: name, type: type).to_a
+  end
+
+  # @return [ModuleItem]
+  def overview_page
```

Comment:
FIXME

Reply:
I think we need a UserTrainingModules object, initialised with the user that holds some of this behaviour.

Reply:
Great minds 🖖🏼 . I was headed that way but was considering `/services/user_training`, WDYT?